ui/imports/shared/status/StatusChatInput.qml

```diff
@@ -1292,14 +1293,16 @@ Rectangle {
 
                             onTextChanged: {
                                 if (length <= control.messageLimit) {
-                                    var symbols = ":='xX><0O;*dB8-D#%\\";
-                                    var symbolIdx = ((cursorPosition > 2) && (symbols.indexOf(getText((cursorPosition - 2), (cursorPosition - 1)))!==-1))
+                                    const symbols = ":='xX><0O;*dB8-D#%\\";
+                                    const symbolIdx = ((cursorPosition > 2) && (symbols.indexOf(getText((cursorPosition - 2), (cursorPosition - 1)))!==-1))
                                                     ? (cursorPosition -1) : (symbols.indexOf(getText(0, 1))!==-1) ? 2 : -1;
                                     if ((getText(symbolIdx-2, (symbolIdx-1)) === " ") || (symbolIdx === 2)) {
+                                        const textBefore1 = getText((symbolIdx-1), (symbolIdx+1))
+                                        const textBefore2 = getText((symbolIdx-2), (symbolIdx+1))
+                                        const has2CharsText = getText((symbolIdx-4), (symbolIdx-2))
                                         const emojis = StatusQUtils.Emoji.emojiJSON.emoji_json.filter(function (emoji) {
```

Comment:
I think we can improve even more with [find](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) instead of `filter`. And return true on the first match.

Reply:
Fixed! Pls check the last commit ;)

Reply:
![image](https://user-images.githubusercontent.com/5377645/230443776-f16d733c-ef14-43cc-bab6-e9d57f15ce84.png)